cli/injector/main.go

```diff
@@ -329,6 +339,50 @@ func injectAndWait(cmd *cobra.Command, args []string) {
 		}
 	}
 
+	if len(pulsingInjectors) > 0 {
+		isInjected := true
+		sleepDuration := pulseActiveDuration
+
+		for { // This loop will inject, wait, clean, wait until a signal is received
+			select { // Quit on signal reception or sleep and injects / cleans the disruptions
+			case sig := <-signals:
+				log.Infow("an exit signal has been received", "signal", sig.String())
+
+				return
+			case <-time.After(sleepDuration):
+				if !isInjected {
+					for _, inj := range pulsingInjectors {
+						if err := backoff.RetryNotify(inj.Inject, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), 3), retryNotifyHandler); err != nil {
+							log.Errorw("pulsing disruption injection failed", "error", err)
+
+							return
+						}
+					}
+
+					log.Info("pulsing disruption(s) are in active mode")
+
+					sleepDuration = pulseActiveDuration
+				} else {
+					for _, inj := range pulsingInjectors {
+						if err := backoff.RetryNotify(inj.Clean, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), 3), retryNotifyHandler); err != nil {
+							log.Errorw("pulsing disruption clean failed", "error", err)
+
+							return
+						}
+					}
+
+					log.Info("pulsing disruption(s) are in dormant mode")
+
+					sleepDuration = pulseDormantDuration
+				}
+
+				isInjected = !isInjected
+			}
+		}
+	} else if pulseActiveDuration > 0 && pulseDormantDuration > 0 && !errOnInject {
```

Comment:
Same as my comment above, not sure that this condition is meaningful if it is prevented by the controller already?